src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -90,6 +92,17 @@
    * accesses to instances map should be protected by this lock.
    */
   private static final Object appsLock = new Object();
+  private static final Supplier<GoogleCredentials> DEFAULT_CREDS_CALLABLE =
+      new Supplier<GoogleCredentials>() {
+        @Override
+        public GoogleCredentials get() {
+          try {
+            return GoogleCredentials.getApplicationDefault();
```

Comment:
I think this needs to be scoped too. Perhaps move this to `FirebaseOptions` with package-level visibility, and use it here as `FirebaseOptions.APPLICATION_DEFAULT_CREDENTIALS`.

Reply:
Done